x/tokenfactory/keeper/bankactions.go

```diff
@@ -0,0 +1,54 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/osmosis-labs/osmosis/v7/x/tokenfactory/types"
+)
+
+func (k Keeper) mintTo(ctx sdk.Context, amount sdk.Coin, mintTo string) error {
```

Comment:
We should add defense in depth validation that amount.Denom is a token factory denom